src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Windows/KeyAgreeRecipientInfoPalWindows.cs

```diff
@@ -96,17 +96,12 @@ public sealed override SubjectIdentifierOrKey OriginatorIdentifierOrKey
                         delegate (CMSG_KEY_AGREE_RECIPIENT_INFO* recipient)
                         {
                             CMsgKeyAgreeOriginatorChoice originatorChoice = recipient->dwOriginatorChoice;
-                            switch (originatorChoice)
+                            return originatorChoice switch
                             {
-                                case CMsgKeyAgreeOriginatorChoice.CMSG_KEY_AGREE_ORIGINATOR_CERT:
-                                    return recipient->OriginatorCertId.ToSubjectIdentifierOrKey();
-
-                                case CMsgKeyAgreeOriginatorChoice.CMSG_KEY_AGREE_ORIGINATOR_PUBLIC_KEY:
-                                    return recipient->OriginatorPublicKeyInfo.ToSubjectIdentifierOrKey();
-
-                                default:
-                                    throw new CryptographicException(SR.Format(SR.Cryptography_Cms_Invalid_Originator_Identifier_Choice, originatorChoice));
-                            }
+                                CMsgKeyAgreeOriginatorChoice.CMSG_KEY_AGREE_ORIGINATOR_CERT => recipient->OriginatorCertId.ToSubjectIdentifierOrKey(),
+                                CMsgKeyAgreeOriginatorChoice.CMSG_KEY_AGREE_ORIGINATOR_PUBLIC_KEY => recipient->OriginatorPublicKeyInfo.ToSubjectIdentifierOrKey(),
```

Comment:
One specific point of contention: I don't like how long it made this line :)

Reply:
> One specific point of contention: I don't like how long it made this line :)

If that's your concern we can easily move everything after the => to the next line :)